src/H5VLnative_datatype.c

```diff
@@ -237,6 +237,12 @@ H5VL__native_datatype_specific(void *obj, H5VL_datatype_specific_args_t *args, h
     switch (args->op_type) {
         /* H5VL_DATATYPE_FLUSH */
         case H5VL_DATATYPE_FLUSH: {
+            /* Currently, H6Oflush causes H5Fclose to trigger an assertion failure in metadata cache.
+             * Leave this situation for the future solution */
+            if (H5FD_MPIO == H5F_get_driver_id(dt->oloc.file))
+                HGOTO_ERROR(H5E_DATASET, H5E_CANTFLUSH, FAIL,
```

Comment:
I'd change the minor code on these to H5E_UNSUPPORTED and just say that H5Oflush isn't supported in parallel in all the error messages rather than saying it doesn't work properly.

Reply:
Fixed.